Lab_13-15/raport.md

```diff
@@ -553,54 +649,235 @@ FROM us_cities c1, us_cities c2
+WHERE a.id = 1 AND b.id = 2;
+```
+
+> Wniosek: Funkcja SDO_UNION zwraca sumę dwóch podanych geometrii. Zwraca nowy obiekt geometryczny, który obejmuje obszar obu wejściowych geometrii.
+
+**sdo_difference**
 
 ```sql
---  ...
+SELECT SDO_GEOM.SDO_DIFFERENCE(a.geom, b.geom, 0.005) AS difference_geom
+FROM geometry_table a, geometry_table b
+WHERE a.id = 1 AND b.id = 2;
 ```
 
+> Wniosek: Funkcja SDO_DIFFERENCE róznice miedzy dwoma podanymi geometriami. Zwraca nowy obiekt geometryczny, który reprezentuje obszar pierwszej geometrii, z którego usunięto obszar drugiej geometrii.
 
-Zadanie 8
+    b.     sdo_buffer
 
-Wykonaj kilka własnych przykładów/analiz
+**sdo_buffer**
 
+```sql
+SELECT SDO_GEOM.SDO_BUFFER(geom, 0.1, 0.005) AS buffer_geom
+FROM geometry_table
+WHERE id = 1;
+```
 
->Wyniki, zrzut ekranu, komentarz
+> Wniosek: Funkcja SDO_BUFFER tworzy bufer wokol podanej geometrii. Zwraca nową geometrię, która reprezentuje obszar w określonej odległości od wejściowej geometrii.
+
+    c.     sdo_centroid, sdo_mbr, sdo_convexhull, sdo_simplify
+
+**sdo_centroid**
 
 ```sql
---  ...
+SELECT SDO_GEOM.SDO_CENTROID(geom, 0.005) AS centroid_geom
+FROM geometry_table
+WHERE id = 1;
+```
+
+> Wniosek: Funkcja SDO_CENTROID oblicza centroid dla podanej geometrii. Zwraca punkt geometryczny, który jest środkiem ciężkości wejściowej geometrii.
+
+**sdo_mbr**
+
+```sql
+SELECT SDO_GEOM.SDO_MBR(geom) AS mbr_geom
+FROM geometry_table
+WHERE id = 1;
+```
+
+> Wniosek: Funkcja SDO_MBR zwraca minimalny zewnętrzny prostokąt dla podanej geometrii. Zwraca prostokątną geometrię, która całkowicie obejmuje wejściowy obiekt geometryczny.
+
+**sdo_convexhull**
+
+```sql
+SELECT SDO_GEOM.SDO_CONVEXHULL(geom, 0.005) AS convexhull_geom
+FROM geometry_table
+WHERE id = 1;
 ```
 
+> Wniosek: Funkcja SDO_CONVEXHULL tworzy otoczkę wypukłą wokół podanej geometrii. Zwraca najmniejszy wypukły wielokąt, który obejmuje wszystkie punkty wejściowej geometrii.
+
+**sdo_simplify**
+
+```sql
+SELECT SDO_GEOM.SDO_SIMPLIFY(geom, 0.005) AS simplified_geom
+FROM geometry_table
+WHERE id = 1;
+```
+
+> Wniosek: Funkcja SDO_SIMPLIFY upraszcza podaną geometrię. Zwraca uproszczoną wersję wejściowej geometrii, zachowującą jej ogólny kształt i cechy, ale z mniejszą liczbą punktów.
+
+f)      Itp. (własne przykłady)
+
+> Wyniki, zrzut ekranu, komentarz (dla każdego z podpunktów)
+
+ - Pola powierzchni stanów od największego do najmniejszego w kilometrach kwadratowych (5 przykładów)
+
+```sql
+SELECT state, SDO_GEOM.SDO_AREA(geom, 0.5,'unit=SQ_KM') as area
+FROM us_states
+ORDER BY area DESC
+```
+
+('Alaska', 1501593.16)
+('Texas', 687005.67)
+('California', 410033.45)
+('Montana', 380815.66)
+('New Mexico', 314906.25)
+
+# Zadanie 8
+
+Wykonaj kilka własnych przykładów/analiz
+
+
+> Wyniki, zrzut ekranu, komentarz
+
+1. Analiza przykładu najdłuższej rzeki przy użyciu narzędzi dostarczanych przez język
+Python (folium, geojson)
+
+**Skrypt w Pythonie**
+
+![](img/ex8/1_s.png)
```

Comment:
Maybe paste the code instead of screenshots in exercise 8? The teacher mentioned that he prefers to copy code from the report

Reply:
I changed screenshots to Python code